poly/src/domain/utils.rs

```diff
@@ -202,7 +203,7 @@ impl<F: FftField> Iterator for Elements<F> {
             let cur_elem = self.cur_elem;
             self.cur_elem *= &self.group_gen;
             self.cur_pow += 1;
-            Some(cur_elem)
+            Some(self.offset * cur_elem)
```

Comment:
We should just set `cur_elem` to be multiplied by `self.offset` at construction time, to avoid the extra multiplication on every call to `next()`